### PR TITLE
Fix type error

### LINKED ISSUE #12974: LocaleProvider use wrong children type
- [ ] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Version
3.10.4

### Environment
Window 7

### Reproduction link
[https://github.com/ant-design/ant-design/blob/488f5a101d8460b7a689c3f2526ae2cba7af8d6f/components/locale-provider/index.tsx#L23](https://github.com/ant-design/ant-design/blob/488f5a101d8460b7a689c3f2526ae2cba7af8d6f/components/locale-provider/index.tsx#L23)

### Steps to reproduce
`LocaleProviderProps` is using `React.ReactElement<any>` instead of `ReactNode` for `children` which cause a type error.

### What is expected?
Use  `ReactNode` for `children`

### What is actually happening?
Use  `React.ReactElement<any>` for `children`

### PR DESCRIPTION
**Pull request by ztplz**
Trying to close #12974.
Here is [Docs](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/7a5dfdb5f6d9cbb438892a0db8c7d195751a984e/types/react/index.d.ts#L2246).

### CODE REVIEW

File: components/locale-provider/index.tsx [hunk 1]
```diff
@@ -20,7 +20,7 @@ export interface Locale {
 
 export interface LocaleProviderProps {
   locale: Locale;
-  children?: React.ReactElement<any>;
+  children?: React.ReactNode;
```
Comment (yesmeck) [inline components/locale-provider/index.tsx]: `React.ReactChilren`

Comment (ztplz) [reply components/locale-provider/index.tsx]: `React.Children.only(this.props.children)` `only` require a parameter what type should be `ReactNode`.

Comment (yesmeck) [reply components/locale-provider/index.tsx]: Hmm, you're right.

**Review by yesmeck — APPROVED** [review]

**Comment by afc163** [comment]
Consider the following situation.

```jsx
<LocaleProvider>
  <div />
  <div />
  <div />
</LocaleProvider>
```

**Comment by ztplz** [comment]
@afc163  If this is the case, these component should be wrapped with a root element according to [Antd-Docs](https://ant.design/components/locale-provider/). Or we can use `React.Children.map` instead of `React.Children.only` to make it work for this situation?

**Comment by joshuaavalon** [comment]
I faced this problem when I was trying to create a wrapper.
```
<LocaleProvider>
{children}
</LocaleProvider>
```

**Comment by ztplz** [comment]
@joshuaavalon This is [implementation](https://github.com/ant-design/ant-design/blob/master/components/locale-provider/index.tsx#L80), currently it only supports one root element.

**Comment by joshuaavalon** [comment]
@ztplz React Router is in the same situation but they still allows it.

**Pull request merged**